src/main/java/games/strategy/engine/data/UnitCollection.java

```diff
@@ -257,4 +219,37 @@ public String toString() {
   public Iterator<Unit> iterator() {
     return Collections.unmodifiableList(m_units).iterator();
   }
+
+  @Override
+  public boolean contains(Object object) {
+    return m_units.contains(object);
+  }
+
+  @Override
+  public Object[] toArray() {
+    return m_units.toArray();
+  }
+
+  @Override
+  public <T> T[] toArray(T[] array) {
+    return m_units.toArray(array);
+  }
+
+  @Override
+  public boolean remove(Object object) {
+    final boolean result = m_units.remove(object);
+    m_holder.notifyChanged();
+    return result;
+  }
+
+  @Override
+  public boolean retainAll(Collection<?> collection) {
+    return m_units.retainAll(collection);
+  }
+
+  @Override
+  public void clear() {
+    m_units.clear();
+    m_holder.notifyChanged();
+  }
```

Comment:
I'm not a huge fan of the various wrapper types for collections of a specific type like IntegerMap or UnitCollection.
While IntegerMap is kinda useful because it adds operations like `add(key)` etc. UnitCollection is essentially a wrapper for an ArrayList of unit types wich notifies pseudo-observers whenever a value is being added or removed